third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -265,37 +264,21 @@ func resourceDataflowJobDelete(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	// Retry updating the state while the job is not ready to be canceled/drained.
-	err = resource.Retry(time.Minute*time.Duration(15), func() *resource.RetryError {
-		// To terminate a dataflow job, we update the job with a requested
-		// terminal state.
-		job := &dataflow.Job{
-			RequestedState: requestedState,
+	job := &dataflow.Job{
+		RequestedState: requestedState,
+	}
+	err = retryTimeDuration(func() (updateErr error) {
+		_, updateErr = resourceDataflowJobUpdateJob(config, project, region, id, job)
+		if isDataflowJobTerminatedError(updateErr) {
+			// Job has already been terminated, skip.
+			return nil
 		}
-
-		_, updateErr := resourceDataflowJobUpdateJob(config, project, region, id, job)
-		if updateErr != nil {
-			gerr, isGoogleErr := err.(*googleapi.Error)
-			if !isGoogleErr {
-				// If we have an error and it's not a google-specific error, we should go ahead and return.
-				return resource.NonRetryableError(err)
-			}
-
-			if strings.Contains(gerr.Message, "not yet ready for canceling") {
-				// Retry cancelling job if it's not ready.
-				// Sleep to avoid hitting update quota with repeated attempts.
-				time.Sleep(5 * time.Second)
-				return resource.RetryableError(err)
-			}
-
-			if strings.Contains(gerr.Message, "Job has terminated") {
-				// Job has already been terminated, skip.
-				return nil
-			}
+		if _, ok := updateErr.(*googleapi.Error); ok {
```

Comment:
i think we can get rid of this block - we should still be handling other googleapi errors, but the retry predicates will handle the appropriate error handling